### DISCUSSION TITLE
missing typescript checking inside store getter and action functions [Nuxt3]

### DISCUSSION THREAD

**Discussion by jamiecarter7** [vuejs/pinia#1575] [Help and Questions] [answered]
### Reproduction

-

### Steps to reproduce the bug

using Nuxt3

### Expected behavior
```
import { defineStore } from "pinia";

export const useOptions = defineStore("options", {
  state: () => ({
    cardShow: false as Boolean,
    keyboard: false as boolean,
  }),
  actions: {
    toggleCard(): void {
      this.cardShow = !this.cardHowToPlay;
      this.keyboardTypo = !this.keyboardTypo;

    },
  },
});
```
hovering 'cardShow' in toggleCard should show it as a 'Boolean'
'this.keyboardTypo' should show an error as it doesn't exist.

### Actual behavior

hovering 'this', 'cardShow' in toggleCard show 'any'
'this.keyboardTypo' should show an error as it does not exist and shows as 'any' when I hover.
<img width="467" alt="Screenshot 2022-08-20 at 13 09 39" src="https://user-images.githubusercontent.com/9541902/185745553-1850a2b0-759e-4a54-bbae-a35cb0bf151d.png">
<img width="484" alt="Screenshot 2022-08-20 at 13 09 28" src="https://user-images.githubusercontent.com/9541902/185745554-01e74b28-566e-4686-bb3c-198bdc6d56f5.png">

### Additional information

The object for actions is fully mapped
The object for the action (function) toggleCard is also correctly mapped.
Using this store in another store shows full mapping.

It seems to be an internal mapping within a single store.

Pinia seems to be unable to map the variable/functions **inside getters and actions functions**

**Comment by posva** [comment, marked as answer]
you need `strict: true` in tsconfig

**Comment by jamiecarter7** [comment]
I had no idea TS was not even enabled on new Nuxt3 projects.

The object types are now showing as expected.

You've saved me hours of my life
